Source/Coop/Session/ExitStatusPatch.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Linq;
+using System.Reflection;
+using EFT;
+using EFT.UI.SessionEnd;
+
+namespace StayInTarkov.Coop.Session
+{
+    public class ExitStatusPatch : ModulePatch
+    {
+        protected override MethodBase GetTargetMethod()
+        {
+            var desiredType = typeof(SessionResultExitStatus);
+            var desiredMethod = ReflectionHelpers.GetAllMethodsForType(desiredType).FirstOrDefault(IsTargetMethod);
+
+            return desiredMethod;
+        }
+
+        private static bool IsTargetMethod(MethodInfo mi)
+        {
+            var parameters = mi.GetParameters();
+            return parameters.Length == 7
+                   && parameters[0].Name == "activeProfile"
+                   && parameters[1].Name == "lastPlayerState"
+                   && parameters[2].Name == "side"
+                   && parameters[3].Name == "exitStatus";
+        }
+
+        [PatchPrefix]
+        private static void PatchPrefix(ref Profile activeProfile, ref ExitStatus exitStatus)
+        {
+            Logger.LogInfo("PatchPrefix");
+            // If extracted alive subtract 300 from session experience to account for extract bonus, if below 200 it's a run through
+            if (exitStatus == ExitStatus.Survived)
+            {
+                var xpGained = activeProfile.EftStats.TotalSessionExperience;
+                if (xpGained > 300) xpGained -= 300;
+                if (xpGained < 200) exitStatus = ExitStatus.Runner;
```

Comment:
Same as "ExperienceGainPatch.cs" changes.

Reply:
Implemented this but not the SurvivedTimeRequirement. It's 420 seconds at which point players get a 300 xp bonus that puts them over the SurvivedExpRequirement so it would be redundant. I also don't know a good way to get past raid time information after the fact.